frontend/src/bundles/common/components/toggle/toggle.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { useFormController } from '~/bundles/common/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+    control: Control<T, null>;
+    label: string;
+    name: FieldPath<T>;
+    type?: 'checkbox' | 'radio' | 'switch';
+};
+
+const Toggle = <T extends FieldValues>({
```

Comment:
I would never have guessed that a toggle component could be a checkbox(or radio button), so it would be better to split this component into 3 different components.

Reply:
Created 3 separate components